internal/biz/resources/resources.go

```diff
@@ -109,23 +111,125 @@ func (uc *Usecase) Create(ctx context.Context, m *model.Resource) (*model.Resour
 
 	if uc.Authz != nil {
 		// Send workspace for the created resource
-		err := biz.DefaultSetWorkspace(ctx, uc.Namespace, m, uc.Authz)
+		ct, err := biz.DefaultSetWorkspace(ctx, uc.Namespace, m, uc.Authz)
 		if err != nil {
 			return nil, err
 		}
+
+		m.ConsistencyToken = ct
+
+		if !uc.DisablePersistence {
+			_, _, err = uc.reporterResourceRepository.Update(ctx, m, m.ID)
+			if err != nil {
+				return nil, err
+			}
+		}
```

Comment:
Nice, thanks for leveraging the DisablePersistence flag. Just a heads up we'll likely be ripping this out with the consistency work so we'll make sure you guys are looped in when this paradigm changes.